### PR TITLE
update high freq demo

### PR DESCRIPTION
**Pull request by javaThonc**
## Description
Create a high-frequency demo with the calculation of precision to calculate the right prediction level and a classification version of lightGBM that is designed for the high-frequency data classification method as well as a workflow that can be used to backtest later.

## Motivation and Context
Create a high freq demo that can be used to test whether a dataset contains signal for trading

## Types of changes
- [ ] Fix bugs
- [x] Add new feature
- [ ] Update documentation

### CODE REVIEW

**Comment by you-n-g** [comment]
@javaThonc 
1. We need more documents about this example 
2. It will be better if we could demonstrate the evaluation metrics about high frequency trading algorithms in the documents  like `https://github.com/microsoft/qlib/tree/main/examples`

File: qlib/contrib/eva/alpha.py [hunk 1]
```diff
@@ -8,6 +8,48 @@
...
+        df.dropna(inplace=True)
+
+    group = df.groupby(level=date_col)
+
+    N = lambda x: int(len(x) * quantile)
+    # find the top/low quantile of prediction and treat them as long and short target
+    long = group.apply(lambda x: x.nlargest(N(x), columns="pred").label).reset_index(level=0, drop=True)
+    short = group.apply(lambda x: x.nsmallest(N(x), columns="pred").label).reset_index(level=0, drop=True)
+
+    groupll = long.groupby(date_col)
```
Comment (you-n-g) [inline qlib/contrib/eva/alpha.py:43]: what does `ll` mean?

File: qlib/contrib/model/highfreq_gdbt_model.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    """LightGBM Model"""
+
+    def __init__(self, loss="mse", **kwargs):
+        if loss not in {"mse", "binary"}:
+            raise NotImplementedError
+        self.params = {"objective": loss, "verbosity": -1}
+        self.params.update(kwargs)
+        self.model = None
+
+    def _cal_signal_metrics(self, y_test, l_cut, r_cut):
```
Comment (you-n-g) [inline qlib/contrib/model/highfreq_gdbt_model.py:24]: Why don't you use `calc_prec` here?

File: qlib/workflow/record_temp.py [hunk 3]
```diff
@@ -156,6 +156,60 @@ def load(self, name="pred.pkl"):
...
+        pprint(metrics)
+
+    def list(self):
+        paths = [
+            self.get_path("ic.pkl"),
+            self.get_path("ric.pkl"),
+            self.get_path("long_pre.pkl"),
+            self.get_path("short_pre.pkl"),
+        ]
+        paths.extend([self.get_path("long_short_r.pkl"), self.get_path("long_avg_r.pkl")])
```
Comment (you-n-g) [inline qlib/workflow/record_temp.py:209]: Why is  `extend` a better choice than writing it in the list directly here?

File: qlib/contrib/eva/alpha.py [hunk 4]
```diff
@@ -8,6 +8,48 @@
...
+    date_col :
+        date_col
+
+    Returns
+    -------
+    (pd.Series, pd.Series)
+        long precision and short precision in time level
+    """
+    if is_alpha:
+        label = label - label.mean(level=0)
```
[the remark this answers is not in the document]
Comment (you-n-g) [reply qlib/contrib/eva/alpha.py:28]: label.mean(level=date_col)

File: examples/highfreq/workflow_config_High_Freq_Tree_Alpha158.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+qlib_init:
+    provider_uri: "~/.qlib/qlib_data/yahoo_cn_1min"
+    region: cn
+market: &market ['SH605222', 'SZ002796', 'SZ002246', 'SZ000713', 'SZ002820', 'SH601328', 'SZ000668', 'SH603359', 'SZ002144', 'SH600195', 'SH603685', 'SH603386', 'SZ002586', 'SZ000573', 'SZ000605', 'SZ002842', 'SH600068', 'SZ300547', 'SZ000926', 'SZ002036', 'SZ002161', 'SH600715', 'SZ300427', 'SZ002573', 'SZ300142', 'SH605116', 'SZ002951', 'SH600276', 'SZ002437', 'SH603355', 'SZ002893', 'SH600584']
```
[the remark this answers is not in the document]
Comment (zhupr) [reply examples/highfreq/workflow_config_High_Freq_Tree_Alpha158.yaml:4]: updated: `python get_data.py qlib_data --name qlib_data --target_dir ~/.qlib/qlib_data/cn_data_1min --interval 1min --region cn`

**Comment by you-n-g** [comment]
@javaThonc   Please remember to update the evaluation results on high frequency data and send us the profiling results.

Besides,  don't forget to update the results of [tabnet](https://github.com/microsoft/qlib/pull/331/files) .

**Pull request merged**